src/gui/systray.h

```diff
@@ -61,6 +61,9 @@ class Systray
     Q_INVOKABLE void setOpened();
     Q_INVOKABLE void setClosed();
     Q_INVOKABLE void positionWindow(QQuickWindow *window) const;
+    Q_INVOKABLE bool hasNotification() const { return !_notificationList.isEmpty(); }
```

Comment:
Either turn that into a Q_PROPERTY or just check the notifications string for emptiness